src/src/lib.zig

```diff
@@ -0,0 +1,480 @@
+    num_files: c_int,
+    cancellable: ?*c.GCancellable,
+    result_callback: AsyncResultCallback(?*c.GList),
+    err: [*c][*c]c.GError,
+) void {
+    var payload = std.heap.c_allocator.create(AsyncPayload(?*c.GList)) catch return;
+    payload.callback = result_callback;
+    payload.err = err;
+
+    c.g_file_enumerator_next_files_async(
+        enumerator,
+        num_files,
+        c.G_PRIORITY_DEFAULT,
+        cancellable,
+        fileEnumNextFilesFinish,
+        @ptrCast(payload),
+    );
+}
+
+pub export fn fileenum_get_file(enumerator: *c.GFileEnumerator, info: *c.GFileInfo) ?*c.GFile {
+    return c.g_file_enumerator_get_child(enumerator, info);
+}
+
+// -------- FILE INFO -------- //
+pub export fn fileinfo_get_name(info: *c.GFileInfo) [*c]const u8 {
+    return c.g_file_info_get_name(info);
+}
+
+pub export fn fileinfo_get_file_type(info: *c.GFileInfo) c_uint {
+    return c.g_file_info_get_file_type(info);
+}
+
+pub export fn fileinfo_get_creation_time(info: *c.GFileInfo) ?[*]const u8 {
+    const time = c.g_file_info_get_creation_date_time(info);
+    if (time) |_time| {
+        return c.g_date_time_format_iso8601(_time);
+    }
+    return null;
+}
+
+pub export fn fileinfo_get_access_time(info: *c.GFileInfo) ?[*]const u8 {
+    const time = c.g_file_info_get_access_date_time(info);
+    if (time) |_time| {
+        return c.g_date_time_format_iso8601(_time);
+    }
+    return null;
+}
+
+pub export fn fileinfo_get_modification_time(info: *c.GFileInfo) ?[*]const u8 {
+    const time = c.g_file_info_get_modification_date_time(info);
+    if (time) |_time| {
+        return c.g_date_time_format_iso8601(_time);
+    }
+    return null;
+}
+
+pub export fn fileinfo_get_size(info: *c.GFileInfo) c_long {
+    return c.g_file_info_get_size(info);
+}
+
+pub export fn fileinfo_get_is_hidden(info: *c.GFileInfo) bool {
+    return c.g_file_info_get_is_hidden(info) == 1;
+}
+
+pub export fn fileinfo_list_attributes(info: *c.GFileInfo, namespace: ?[*]u8) [*c][*c]u8 {
+    return c.g_file_info_list_attributes(info, namespace);
+}
+
+// -------- CANCELLABLE -------- //
+fn onCancel(_: [*c]c.GCancellable, user_data: ?*anyopaque) callconv(.c) void {
+    const callback: *fn () callconv(.c) void = @ptrCast(user_data);
+    callback();
+}
+
+pub export fn cancellable_new(
+    on_cancel: ?*fn () void,
+) *c.GCancellable {
+    const cancellable = c.g_cancellable_new();
+    if (on_cancel) |callback| {
+        _ = c.g_cancellable_connect(
+            cancellable,
+            c.G_CALLBACK(onCancel),
+            @ptrCast(callback),
+            null,
+        );
+    }
+    return cancellable;
+}
+
+pub export fn cancellable_cancel(cancellable: *c.GCancellable) void {
+    c.g_cancellable_cancel(cancellable);
+}
+
+pub export fn cancellable_is_cancelled(cancellable: *c.GCancellable) bool {
+    return c.g_cancellable_is_cancelled(cancellable) == 1;
+}
+
+pub export fn cancellable_destroy(cancellable: *c.GCancellable) void {
```

Comment:
we should also disconnect eventual on cancel callbacks, don't know if it's required but better not to mess with native memory too much. will read glib docs before deciding